app/controllers/reviews_controller.rb

```diff
@@ -45,7 +37,8 @@ def attempt_create(review)
       if review.save
         redirect_to "/shelters/#{params[:shelter_id]}"
       else
-        flash[:notice] = "You need to fill in a title, rating, and content in order to submit a shelter review"
+        @shelter_id = params[:shelter_id]
+        flash.now[:notice] = "You need to fill in a title, rating, and content in order to submit a shelter review"
```

Comment:
before the error was persisting even after making a valid review, flash.now allows it to disappear when it needs to